static/js/stream_edit.js

```diff
@@ -676,6 +676,15 @@ exports.initialize = function () {
         }
 
         subs.sub_or_unsub(sub);
+
+        // If the user click on the Subscribe button in nonsubbed_stream_narrow_message, change the banner to empty_narrow_message from nonsubbed_stream_narrow_message.
+        if (
+            $(".stream_sub_unsub_button").closest("[id]").attr("id") ===
+            "nonsubbed_stream_narrow_message"
```

Comment:
You can just put `$(".stream_sub_unsub_button").closest("#nonsubbed_stream_narrow_message").length` in the if condition and it will work fine.